recipe_scrapers/americastestkitchen.py

```diff
@@ -0,0 +1,90 @@
+# mypy: allow-untyped-defs
+
+import functools
+import json
+import re
+
+from recipe_scrapers._grouping_utils import IngredientGroup
+
+from ._abstract import AbstractScraper
+from ._utils import get_minutes, normalize_string
+
+
+class AmericasTestKitchen(AbstractScraper):
+
+    @classmethod
+    def host(cls, domain="americastestkitchen.com"):
+        return domain
+
+    def author(self):
+        return self.schema.author()
+
+    def title(self):
+        return self.schema.title()
+
+    def description(self):
+        return self.schema.description()
+
+    def total_time(self):
+        return get_minutes(self._get_additional_details.get("recipeTimeNote"))
+
+    def image(self):
+        return self.schema.image()
+
+    def ingredients(self):
+        ingredients = []
+        for group in self._get_additional_details.get("ingredientGroups"):
+            ingredients += group["fields"]["recipeIngredientItems"]
+        return [
+            "{} {} {} {}".format(
+                i["fields"]["qty"] or "",
+                i["fields"]["measurement"] or "",
+                i["fields"]["ingredient"]["fields"]["title"] or "",
+                i["fields"]["postText"] or "",
+            )
+            for i in ingredients
+        ]
+
+    def ingredient_groups(self):
+        ingredient_groups = []
+        for group in self._get_additional_details.get("ingredientGroups"):
+            ingredients = [
+                "{} {} {} {}".format(
+                    i["fields"]["qty"] or "",
+                    i["fields"]["measurement"] or "",
+                    i["fields"]["ingredient"]["fields"]["title"] or "",
+                    i["fields"]["postText"] or "",
+                )
+                for i in group["fields"]["recipeIngredientItems"]
+            ]
+            purpose = group["fields"]["title"]
+            ingredient_groups.append(
+                IngredientGroup(ingredients=ingredients, purpose=purpose)
+            )
+        return ingredient_groups
```

Comment:
Note: this suggestion also relates to some of the previous `postText` discussion:

I think it could be worth extracting some of the common logic between these two methods into a re-usable utility.  I've experiemented with something like:

```python
    @staticmethod
    def _parse_ingredient_item(ingredient_item):
        fields = ingredient_item["fields"]
        fragments = (
            fields["qty"] or "",
            fields["measurement"] or "",
            fields["ingredient"]["fields"]["title"] or "",
            fields["postText"] or "",
        )
        return " ".join(fragment.rstrip() for fragment in fragments).rstrip()
```

Reply:
better - but that still leaves some ingredients with spaces before a comma like "1/2  small onion , roughly chopped"

Is it worthing adding a `.replace(" ,", ",")` ?

Reply:
> better - but that still leaves some ingredients with spaces before a comma like "1/2 small onion , roughly chopped"
> 
> Is it worthing adding a `.replace(" ,", ",")` ?

That suggestion seems good to me, yep :+1: